spec/custom-provider-spec.js

```diff
@@ -3,23 +3,48 @@
 import fs from 'fs-extra';
 import temp from 'temp';
 import CustomFile from '../lib/atom-build.js';
+import os from 'os';
 
 describe('custom provider', () => {
   let builder;
+  let createdHomeDir;
   let directory = null;
+  const originalHomedirFn = os.homedir;
 
   temp.track();
 
   beforeEach(() => {
+    createdHomeDir = temp.mkdirSync('atom-build-spec-home');
     directory = fs.realpathSync(temp.mkdirSync({ prefix: 'atom-build-spec-' })) + '/';
     atom.project.setPaths([ directory ]);
+    os.homedir = () => createdHomeDir;
     builder = new CustomFile(directory);
   });
 
   afterEach(() => {
+    os.homedir = originalHomedirFn;
     fs.removeSync(directory);
   });
 
+  it('when there is no .atom-build config file in any elegible directory', () => {
+    expect(builder.isEligible()).toEqual(false);
+  });
+
+  describe('when .atom-build config is on home directory', () => {
+    it('should find json file in home directory', () => {
+      fs.writeFileSync(createdHomeDir + '/.atom-build.json', fs.readFileSync(__dirname + '/fixture/.atom-build.json'));
+      expect(builder.isEligible()).toEqual(true);
+    });
+    it('should find cson file in home directory', () => {
+      fs.writeFileSync(createdHomeDir + '/.atom-build.cson', fs.readFileSync(__dirname + '/fixture/.atom-build.cson'));
+      expect(builder.isEligible()).toEqual(true);
+    });
+    it('should find yml file in home directory', () => {
+      fs.writeFileSync(createdHomeDir + '/.atom-build.yml', fs.readFileSync(__dirname + '/fixture/.atom-build.yml'));
+      expect(builder.isEligible()).toEqual(true);
+    });
+  });
```

Comment:
Absolutely. A separate test to make sure it parses targets. I don't think it is necessary to test all options and file variations, just one file and make sure it gets the target from there should do it.

This looks great and works well from my manual testing! I'd be happy to merge this when you add a test like that!